src/callbacks_step/analysis_dg2d_parallel.jl

```diff
@@ -161,7 +162,7 @@ function integrate_via_indices(func::Func, u,
                             normalize = normalize)
 
     # OBS! Global results are only calculated on MPI root, all other domains receive `nothing`
-    global_integral = MPI.Reduce!(Ref(local_integral), +, mpi_root(), mpi_comm())
+    global_integral = MPI.Reduce!(Ref(local_integral), reduce_vector_plus, mpi_root(), mpi_comm())
```

Comment:
This is the place where we need the vector reduction. Currently, `local_integral` can be a `Float64` in some cases (when we compute the total `entropy`) or an `SVector` (when we compute the total mass of all conserved quantities). What I'm suggesting is to reduce `collect(local_integral)` instead of `Ref(local_integral)`. That should work, shouldn't it?

Reply:
Yeah, that should work, but of course it would require an extra allocation.

Reply:
That's true. I'm just looking for a solution that is at the Pareto front of optimality in terms of code complexity, code generality, and efficiency. While the `@Op` approach is likely best in terms of efficiency, I have some doubts about the code complexity and generality - shall we do it for `SVector{N, T}` for `N in 1:10` (or mre?) and `T in (Float32, Float64)` - and maybe also scalars? Will we need something else? It's kind of bad that Trixi.jl shall be a library and not a single code for a specific application.

Reply:
It's annoying that MPI doesn't specify a "reverse" translation of `MPI_Datatype`.
We could maybe have a dictonary where we do `MPI_Datatype => Type` and then we can use that to get a concrete type, but that would cause a dynamic dispatch...

Reply:
Turns out MPI.jl has support for reverse translations.

I just pushed a commit that allows for `@Op(+, Any)`.

Reply:
Nice! Can we please test this here (Benedict)?

Reply:
Doing this here now: #2066